x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/DataFrameAnalyticsConfig.java

```diff
@@ -135,29 +217,49 @@ public static String documentId(String id) {
         private String source;
         private String dest;
         private List<DataFrameAnalysisConfig> analyses;
+        private Map<String, Object> query;
 
         public String getId() {
             return id;
         }
 
-        public void setId(String id) {
+        public Builder setId(String id) {
             this.id = ExceptionsHelper.requireNonNull(id, ID);
+            return this;
         }
 
-        public void setSource(String source) {
+        public Builder setSource(String source) {
             this.source = ExceptionsHelper.requireNonNull(source, SOURCE);
+            return this;
         }
 
-        public void setDest(String dest) {
+        public Builder setDest(String dest) {
             this.dest = ExceptionsHelper.requireNonNull(dest, DEST);
+            return this;
         }
 
-        public void setAnalyses(List<DataFrameAnalysisConfig> analyses) {
+        public Builder setAnalyses(List<DataFrameAnalysisConfig> analyses) {
             this.analyses = ExceptionsHelper.requireNonNull(analyses, ANALYSES);
+            return this;
+        }
+
+        public Builder setParsedQuery(QueryBuilder query) {
+            try {
+                setQuery(QUERY_TRANSFORMER.toMap(query));
```

Comment:
this approach has a problem: The queryparser injects some defaults into the query, so if you parse it first and than turn it back into a map you get the defaults in the configuration which the user hasn't entered.

For this reason https://github.com/elastic/elasticsearch/pull/37827 preserves the original form even for strict parsing

Reply:
When the query is ran, the defaults are added regardless. Is it better to show the user all the defaults that are used or simply let the underlying system add them at run time?

Reply:
I think we should store the query in the surface form in the config to avoid tight coupling. Runtime adds whatever defaults it wants to add. Defaults might change between versions and/or they could get deprecated. I am more scared about the upgrade path than within 1 version.